static/app/views/detectors/components/forms/metric/visualize.tsx

```diff
@@ -93,18 +88,50 @@ function renderTag(kind: FieldValueKind): React.ReactNode {
   return <Tag type={tagType}>{text}</Tag>;
 }
 
+/**
+ * Aggregate options not allowed for the logs dataset
+ */
 const LOGS_NOT_ALLOWED_AGGREGATES = [
   AggregationKey.FAILURE_RATE,
   AggregationKey.FAILURE_COUNT,
+  AggregationKey.APDEX,
 ];
 
+/**
+ * Additional aggregate options for the spans dataset
+ */
+const EXTRA_AGGREGATES = [AggregationKey.APDEX];
+
+/**
+ * Locks the options because they usually need to count something specific
+ */
+const LOCKED_SPAN_AGGREGATES = {
+  [AggregationKey.APDEX]: {
+    value: DEFAULT_VISUALIZATION_FIELD,
+    label: t('span.duration'),
```

Comment:
might be cleaner if you just set `span.duration` to be the single allowable type for this parameter